osbs/build/build_response.py

```diff
@@ -106,17 +107,39 @@ def get_error_message(self):
         """
         Return an error message based on atomic-reactor's metadata
         """
-        try:
-            str_metadata = graceful_chain_get(self.get_annotations_or_labels(), "plugins-metadata")
-            metadata_dict = json.loads(str_metadata)
-            plugin, error_message = list(metadata_dict['errors'].items())[0]
+        error_reason = self.get_error_reason()
+        if error_reason:
+            error_message = error_reason.get('pod') or None
+            if error_message:
+                return "Error in pod: %s" % error_message
+            plugin = error_reason.get('plugin')[0] or None
+            error_message = error_reason.get('plugin')[1] or None
             if error_message:
                 # Plugin has non-empty error description
                 return "Error in plugin %s: %s" % (plugin, error_message)
             else:
                 return "Error in plugin %s" % plugin
-        except Exception:
-            return None
+
+    def get_error_reason(self):
+        try:
+            str_metadata = graceful_chain_get(self.get_annotations_or_labels(), "plugins-metadata")
+            metadata_dict = json.loads(str_metadata)
+            plugin, error_message = list(metadata_dict['errors'].items())[0]
+            return {'plugin': [plugin, error_message]}
+        except (ValueError, AttributeError):
+            try:
+                from osbs.api import OSBS
+                from osbs.conf import Configuration
+
+                osbs_conf = Configuration(self.get_build_json_dir())
```

Comment:
Creating new `Configuration` and `OSBS` objects in `BuildResponse` seems to blur the layers of abstraction. Does it makes sense to move get-error-reason-for-build logic to `api.py`?

Reply:
I agree about breaking the abstraction, but moving it up to api.py means a change to osbs-client to add the new API call, and to koji_containerbuild and atomic-reactor to use the new API call.

While breaking the abstraction means that only osbs-client needs to change - though atomic-reactor can be changed to take advantage of the new interface.

Reply:
The only reason I'm creating an OSBS object is get a pod_response. Would it be cleaner to pass the pod into the BuildResponse at creation?

Reply:
Conceptually cleaner; a lot uglier in the test cases. Oh, well.

Reply:
I think it makes sense.